plugins/postcss-cascade-layers/src/index.ts

```diff
@@ -14,13 +14,47 @@ function postcssCascadeLayers(opts) {
 					atRule.params = `anon${layerCount}`;
 				}
 
+				let hasNestedLayers = false;
+				let hasUnlayeredStyles = false;
+
+				// check for where a layer has nested layers AND styles outside of those layers
+				atRule.each((node) => {
+					if (node.type == "atrule") {
+						hasNestedLayers = true;
+					} else if (node.type == "rule") {
+						hasUnlayeredStyles = true;
+					}
+				});
+
+				if (hasNestedLayers && hasUnlayeredStyles) {
+					let cloned = [];
+					//create final layer
+					atRule.append({
+						name: `layer ${atRule.params}-last `,
```

Comment:
This creates an `at-rule` with name `layer foo-last`.

The space in between `layer` and `foo-last` makes it seem as it would be an `at-rule` with name `layer` and params `foo-last`, but this will not be the case.

------

This style of node creation will also break sourcemaps.

To avoid broken source maps you can either clone an existing and relevant node or create a completely new node and set the `source` field. ([docs on `source`](https://postcss.org/api/#atrule-source))

I often chose to clone the current node and then remove all child nodes from it. Then I know it will be linked in sourcemaps to the node that triggered the transform.

Reply:
@romainmenke thanks for this context! I was wondering why I would get the `broken source map` error and now that makes sense! I think I was thrown off a bit by the append example in the docs for charset rules ```
root.append({ name: 'charset', params: '"UTF-8"' })``` and tried following it for at-rules. Appreciate your help with this! 🙏🏼

Reply:
If error messages from the test suite are unclear or unhelpful you can always let us know.

We tend to know a bit too much about all this to really make this open and accessible.
Feedback and suggestions are very much appreciated!

Reply:
@sanajaved7 Is this better?
https://github.com/csstools/postcss-plugins/pull/315

Reply:
@romainmenke this is great, thank you!! 🙏🏼 The only other suggestion I have is that guidance somewhere in the docs on inserting nodes and the difference between the various methods like `append()` and `clone()` would be very helpful. 

For me personally, it would have been helpful to see the context you shared on `source` and `clone()` as an alternative to `append()`. Not sure where exactly, maybe  near the append docs as an aside [here](https://postcss.org/api/#atrule-append)? Apologies if this already exists elsewhere and I missed it!

Reply:
Just remembered that there are also plugin guidelines :
https://github.com/postcss/postcss/blob/main/docs/guidelines/plugin.md#24-set-nodesource-for-new-nodes

We can ask PostCSS if it makes sense to add a link to the guidelines from the API docs?
It makes sense to me to keep the docs a bit terse and maybe this is fine if you can also easily find the guidelines?

_added a link to the guidelines in the error message_